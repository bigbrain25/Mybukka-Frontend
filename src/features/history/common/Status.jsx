import React from 'react';
import PropTypes from 'prop-types';

const Status = ({ status, time }) => {
  const colorObj = {
    pending: 'warning',
    delivered: 'success',
    cancelled: 'danger'
  };

  return (
    <div className="text-uppercase">
      <h5 className={`font-size-14 text-${colorObj[status]}`}>{status}</h5>
      <h5 className="light font-size-14 suggested-item-price">{time}</h5>
    </div>
  );
};

export default Status;

Status.propTypes = {
  status: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired
};
