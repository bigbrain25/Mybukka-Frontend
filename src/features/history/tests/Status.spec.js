import React from 'react';
import Status from '../common/Status';

describe('Status component', () => {
  const props = {
    status: 'pending',
    time: '27/3/2019 12:50 am',
  };

  const { rerender, container } = render(<Status {...props} />);

  it('renders properly', () => {
    expect(container).toMatchSnapshot();
  });

  it('re renders properly when delivered status', () => {
    const newProps = {
      ...props,
      status: 'delivered',
    };
    rerender(<Status {...newProps} />);
    expect(container).toMatchSnapshot();
  });

  it('re renders properly when rejected status', () => {
    const newProps = {
      ...props,
      status: 'cancelled',
    };
    rerender(<Status {...newProps} />);
    expect(container).toMatchSnapshot();
  });
});
