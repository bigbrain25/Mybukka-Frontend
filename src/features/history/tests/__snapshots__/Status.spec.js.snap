// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component re renders properly when delivered status 1`] = `
<div>
  <div
    class="text-uppercase"
  >
    <h5
      class="font-size-14 text-success"
    >
      delivered
    </h5>
    <h5
      class="light font-size-13 suggested-item-price"
    >
      27/3/2019 12:50 am
    </h5>
  </div>
</div>
`;

exports[`Status component re renders properly when rejected status 1`] = `
<div>
  <div
    class="text-uppercase"
  >
    <h5
      class="font-size-14 text-danger"
    >
      cancelled
    </h5>
    <h5
      class="light font-size-13 suggested-item-price"
    >
      27/3/2019 12:50 am
    </h5>
  </div>
</div>
`;

exports[`Status component renders properly 1`] = `
<div>
  <div
    class="text-uppercase"
  >
    <h5
      class="font-size-14 text-warning"
    >
      pending
    </h5>
    <h5
      class="light font-size-13 suggested-item-price"
    >
      27/3/2019 12:50 am
    </h5>
  </div>
</div>
`;
